WordPress/src/main/java/org/wordpress/android/ui/prefs/timezone/SiteSettingsTimezoneViewModel.kt

```diff
@@ -0,0 +1,211 @@
+    val timezoneSearch: LiveData<List<TimezonesList>> = Transformations.switchMap(searchInput) { term ->
+        filterTimezones(term)
+    }
+
+    private val _suggestedTimezone = MutableLiveData<String>()
+    val suggestedTimezone = _suggestedTimezone
+
+    private val _timezones = MutableLiveData<List<TimezonesList>>()
+    val timezones = _timezones
+
+
+    fun searchTimezones(query: CharSequence?) {
+        searchInput.value = query.toString()
+    }
+
+    private fun filterTimezones(query: String): LiveData<List<TimezonesList>> {
+        val filteredTimezones = MutableLiveData<List<TimezonesList>>()
+
+        timezonesList.filter { timezone ->
+            when (timezone) {
+                is TimezoneItem -> {
+                    timezone.label.contains(query, true) or timezone.offset.contains(query, true)
+                }
+                else -> false
+            }
+        }.also {
+            filteredTimezones.value = it
+        }
+
+        return filteredTimezones
+    }
+
+    fun onSearchCancelled() {
+        _timezones.postValue(timezonesList)
+    }
+
+    fun getTimezones(context: Activity) {
+        _suggestedTimezone.postValue(ZoneId.systemDefault().id)
+
+        requestTimezones(context)
+    }
+
+    private fun requestTimezones(context: Activity) {
+        val listener = Response.Listener { response: String? ->
+            AppLog.d(SETTINGS, "timezones requested")
+            _showProgress.postValue(false)
+
+            if (!TextUtils.isEmpty(response)) {
+                timezonesList.clear()
+                loadTimezones(response)
+            } else {
+                AppLog.w(SETTINGS, "empty response requesting timezones")
+                _dismiss.postValue(Unit)
+            }
+        }
+
+        val errorListener = Response.ErrorListener { error: VolleyError? ->
+            AppLog.e(SETTINGS, "Error requesting timezones", error)
+            _dismiss.postValue(Unit)
+        }
+
+        val request: StringRequest = object : StringRequest(Constants.URL_TIMEZONE_ENDPOINT, listener, errorListener) {
+            override fun getParams(): Map<String, String> {
+                return RestClientUtils.getRestLocaleParams(context)
+            }
+        }
+
+        _showProgress.postValue(true)
+        val queue = Volley.newRequestQueue(context)
+        queue.add(request)
+    }
+
+    private fun loadTimezones(responseJson: String?) {
+        try {
+            val jsonResponse = JSONObject(responseJson.orEmpty())
+            val jsonTimezonesByContinent = jsonResponse.getJSONObject("timezones_by_continent")
+            val jsonTimezonesManualOffsets = jsonResponse.getJSONArray("manual_utc_offsets")
+
+            Continents.values().map {
+                timezonesList.add(TimezoneHeader(it.s))
+                addTimezoneItems(jsonTimezonesByContinent.getJSONArray(it.s))
+            }
+
+            timezonesList.add(TimezoneHeader("Manual Offsets"))
```

Comment:
Will this string be visible? If so, we would probably like to add it to `strings.xml`